specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json

```
@@ -0,0 +1,248 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "SipRouting Service",
+    "description": "SipRouting Service",
+    "contact": {
+      "email": "acsdevexdisc@microsoft.com"
+    },
+    "version": "2023-03-01"
+  },
+  "paths": {
+    "/sip": {
+      "get": {
+        "tags": [
+          "SipConfiguration"
+        ],
+        "summary": "Gets SIP configuration for resource.",
+        "operationId": "SipRouting_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok.",
+            "schema": {
+              "$ref": "#/definitions/SipConfiguration"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2022-07-13/common.json#/definitions/CommunicationErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get SIP configuration for resource": {
+            "$ref": "./examples/GetSipConfiguration.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "SipConfiguration"
+        ],
+        "summary": "Patches SIP configuration for resource.",
+        "operationId": "SipRouting_Patch",
```

The convention in Azure is to use "update" rather than "patch" in the operationId, which is converted to a method name in generated clients. Please consider
```suggestion
        "operationId": "SipRouting_Update",
```

---

There seems no PUT. So it is CreateOrUpdate?

---

Hi @weidongxu-microsoft , by default, every resource has an empty SIP configuration (we never return 404, but an empty configuration like {trunks:{}, routes:[]}). It's effectively Update vs Patch and as mentioned in my comment, on the previous reviews with the board we agreed on Patch.

---

@dzeliar Normally we defer to conventions used in the service, but it appears that the convention in the Communications service is to use "update" (or in the case of one sub-service "upset") and NOT "patch".  Here's what I found in a survey of the REST APIs in the Communications service:

- operationId for patch on /chat/threads/{chatThreadId}/messages/{chatMessageId} is "ChatThread_UpdateChatMessage"
- operationId for patch on /chat/threads/{chatThreadId} is "ChatThread_UpdateChatThreadProperties"
- operationId for patch on /identities/{id} is "CommunicationIdentity_Update"
- operationId for patch on /mediaCompositions/{mediaCompositionId} is "MediaComposition_Update"
- operationId for patch on /operatorconnect/consents/{operatorId} is "CreateOrUpdateConsent"
- operationId for patch on /phoneNumbers/{phoneNumber}/capabilities is "PhoneNumbers_UpdateCapabilities"
- multiple patch operations in communicationservicejobrouter.json with operationIds that contain "upsert" (but not "patch")

If you use "patch" in your operationId, you will not only break with the Azure conventions but also the conventions used within your own service.

---

Yes, agree with Mike. For Azure convention, `Update` is the merge (else it would be called `CreateOrUpdate`, or `CreateOrReplace`).

---

Hello @mikekistler , @weidongxu-microsoft . I've updated operationId and naming of the dependent objects to Update (please check the[ latest commit](https://github.com/Azure/azure-rest-api-specs/pull/22747/commits/404b10e55294cd8e2d58d153a87f5d30b4dd4f37)) Please check again if it looks correct so we can get a final sign on this API. Thank you